apps/gdaldem_lib.cpp

```diff
@@ -3763,246 +4199,134 @@ GDALDatasetH GDALDEMProcessing(const char *pszDest, GDALDatasetH hSrcDataset,
         {
-            psOptions->eColorSelectionMode = COLOR_SELECTION_NEAREST_ENTRY;
+            throw std::invalid_argument(
+                "-igor and -alt cannot be used together");
         }
-        else if ((EQUAL(papszArgv[i], "--s") || EQUAL(papszArgv[i], "-s") ||
-                  EQUAL(papszArgv[i], "--scale") ||
-                  EQUAL(papszArgv[i], "-scale")) &&
-                 i + 1 < argc)
+
+        if (psOptionsForBinary && aosArgv.size() > 1)
         {
-            ++i;
-            if (!ArgIsNumeric(papszArgv[i]))
-            {
-                CPLError(CE_Failure, CPLE_IllegalArg,
-                         "Numeric value expected for %s", papszArgv[i - 1]);
-                GDALDEMProcessingOptionsFree(psOptions);
-                return nullptr;
-            }
-            psOptions->scale = CPLAtof(papszArgv[i]);
+            psOptionsForBinary->osProcessing = aosArgv[0];
         }
-        else if ((EQUAL(papszArgv[i], "--az") || EQUAL(papszArgv[i], "-az") ||
-                  EQUAL(papszArgv[i], "--azimuth") ||
-                  EQUAL(papszArgv[i], "-azimuth")) &&
-                 i + 1 < argc)
+    };
+
+    try
+    {
+
+        // Ugly hack: papszArgv may not contain the processing mode if coming from
+        // SWIG we have not other option than to check them all
+        const static std::list<std::string> processingModes{
+            {"hillshade", "slope", "aspect", "color-relief", "TRI", "TPI",
+             "roughness"}};
+        if (bNoProcessingMode)
         {
-            ++i;
-            if (!ArgIsNumeric(papszArgv[i]))
+            CPLStringList aosArgvCopy{aosArgv};
+            bool bSuccess = false;
+            for (const auto &processingMode : processingModes)
             {
-                CPLError(CE_Failure, CPLE_IllegalArg,
-                         "Numeric value expected for %s", papszArgv[i - 1]);
-                GDALDEMProcessingOptionsFree(psOptions);
-                return nullptr;
+                aosArgv = aosArgvCopy;
+                aosArgv.InsertString(0, processingMode.c_str());
+                try
+                {
+                    tryHandleArgv();
+                    bSuccess = true;
+                    break;
+                }
+                catch (std::runtime_error &e)
+                {
+                    CPLDebug("GDAL", "Error for mode %s: %s",
+                             processingMode.c_str(), e.what());
+                    continue;
+                }
+                catch (std::invalid_argument &e)
+                {
+                    CPLDebug("GDAL", "Error for mode %s: %s",
+                             processingMode.c_str(), e.what());
+                    continue;
+                }
+                catch (std::logic_error &e)
+                {
+                    CPLDebug("GDAL", "Error for mode %s: %s",
+                             processingMode.c_str(), e.what());
+                    continue;
+                }
```

Comment:
I would not emit those debug messages, as they are "expected" errors

Reply:
done